tensornetwork/contractors/opt_einsum_paths/path_contractors_node_test.py

```diff
@@ -96,3 +96,28 @@ def __call__(self, inputs, output, size_dict, memory_limit=None):
   optimizer = PathOptimizer()
   final_node = path_contractors.custom(nodes, optimizer)
   np.testing.assert_allclose(final_node.tensor, np.ones(5) * 2.0)
+
+
+def test_subgraph_contraction(backend, path_algorithm):
+  a = tn.Node(np.ones((2, 2)), backend=backend)
+  b = tn.Node(np.ones((2, 2)), backend=backend)
+  c = tn.Node(np.ones((2, 2)), backend=backend)
```

Comment:
This test would be more robust if the tensors were more interesting, e.g.

```python
a = 10 + tn.Node(np.arange(4).reshape((2, 2)), backend=backend)
a = 20 + tn.Node(np.arange(4).reshape((2, 2)), backend=backend)
```

The present form of the tensors allows certain bugs to pass through unnoticed, e.g. if we were inadvertently contracting b with c (instead of a with b).

Reply:
Added